main.go

```diff
@@ -325,6 +327,8 @@ func main() {
 	createCmd.MarkFlagRequired("key")
 
 	rootCmd.AddCommand(deleteCmd)
+	deleteCmd.Flags().BoolVar(&permanentDelete, "permanently", false,
```

Comment:
I think we should use `--all-versions` or something more descriptive here instead.

Reply:
Yeah I wasn't so sure about this. GCS uses versioning for both deleted objects and updated objects. So I'm making this distinction: "if the live version of an object is gone, consider that secret deleted". In other words, Berglas only works with "live secrets" (for the most part - see side note below).

That implies that list command's response shouldn't include any secret that has been deleted (that's the change to list.go in this PR). Now, when deleting secrets, deleting old generations from GCS means you can't recover the secrets any more. `--all-generations` doesn't indicate this permanent nature of this operation.

Side note: However, `access deleted-secret --generation gen` command would still return it (if you knew the generation to fetch), and this is one way to recover the secret. In the future, there could be a `berglas recover SECRET` operation that makes this easier.

Reply:
I don't think we need to support deleting a specific generation. If a secret is deleted, we should purge that GCS object and any of its versions. Therefore I don't think we need this flag at all and should just change the semantics of delete now that we have versioning.

Reply:
Yes, that's a valid option too. As a user, you should be able to recover these secrets from elsewhere..

Reply:
Yup. Also, that was the existing contract before we introduced versioning, so we have broken anything.